rfqmsg/messages.go

```diff
@@ -216,3 +216,54 @@ func NewBigIntFixedPoint(coefficient uint64, scale uint8) BigIntFixedPoint {
 		Scale:       scale,
 	}
 }
+
+// TlvFixedPoint is a fixed-point that can be TLV encode/decode.
+type TlvFixedPoint struct {
+	// fp is the underlying BigInt fixed-point.
+	fp BigIntFixedPoint
+}
+
+// NewTlvFixedPointFromBigInt creates a new fixed-point given a BigInt
+// fixed-point.
+func NewTlvFixedPointFromBigInt(fp BigIntFixedPoint) TlvFixedPoint {
+	return TlvFixedPoint{
+		fp: fp,
+	}
+}
+
+// NewTlvFixedPointFromUint64 creates a new fixed point record given a `uint64`
+// coefficient and scale.
+func NewTlvFixedPointFromUint64(coefficient uint64, scale uint8) TlvFixedPoint {
```

Comment:
I wonder why we even need this additional type? Can't we just add the `Record` method to the `BigIntFixedPoint` type?

Reply:
Can't define methods on:
```
type BigIntFixedPoint = FixedPoint[BigInt]
```
Got error:
> cannot define new methods on instantiated type BigIntFixedPoint

It's an alias because I couldn't get `UnitsToMilliSatoshi` to work with `type BigIntFixedPoint FixedPoint[BigInt]`.

However! `UnitsToMilliSatoshi` doesn't need to be generic, we only need it to work with `BigIntFixedPoint`. Which means `BigIntFixedPoint` could become a proper type and we can add `Record` to it.

Making `UnitsToMilliSatoshi` non-generic involves changing many tests. So I've left that for future work.

I agree that `TlvFixedPoint` can be deleted in favour of `BigIntFixedPoint` but perhaps in a future PR. Won't be a breaking change anyway.